src/components/radio-button/index.tsx

```diff
@@ -0,0 +1,28 @@
+import { Component, h, Host, Listen, Prop } from "@stencil/core"
+import { Selected } from "./Selected"
+
+@Component({
+	tag: "smoothly-radio-button",
+	styleUrl: "style.css",
+	scoped: true,
+})
+export class SmoothlyRadioButton {
+	@Prop({ mutable: true }) value?: any
+	active?: Selected
+
+	@Listen("radioSelect")
+	radioSelectHandler(event: CustomEvent<Selected>) {
+		this.active?.select(false)
+		this.active = event.detail
+		this.value = this.active.selected ? event.detail.value : undefined
```

Comment:
is it not cleaner to write either `this.value = event.detail.selected ? event.detail.value : undefined` or `this.value = this.active.selected ? this.active.value : undefined` ? The later one is maybe preferable since no other reference to `event.detail` is done other than the assignment and here.